### PR TITLE
syntax fix for terraform 0.13 ->

### LINKED ISSUE #1: Update script syntax for Terraform >0.13
Terraform gives syntax warnings. Update to newer syntax.

```
Warning: Version constraints inside provider configuration blocks are deprecated

  on main.tf line 4, in provider "azurerm":
   4:     version = "~>2.0"

Terraform 0.13 and earlier allowed provider version constraints inside the
provider configuration block, but that is now deprecated and will be removed
in a future version of Terraform. To silence this warning, move the provider
version constraint into the required_providers block.
```

### PR DESCRIPTION
**Pull request by laavamara**
Fix terraform syntax for versions -> 0.13
closes #1
Signed-off-by: Sami <sami.wickstrom@protonmail.com>

### CODE REVIEW

**Comment by laavamara** [comment]
#9

**Comment by laavamara** [comment]
#1

**Comment by laavamara** [comment]
#1

**Comment by hjhsalo** [comment]
Did you receive similar syntax warning about this code https://github.com/hjhsalo/smad-terraform-azure/blob/master/modules/k8s/main.tf#L1-L6

It seems to be similar / same as the one you fixed in 'main.tf' '/modules/container_registry/main.tf'.

File: main.tf [hunk 1]
```diff
@@ -1,10 +1,17 @@
-provider "azurerm" {
-    # The "feature" block is required for AzureRM provider 2.x. 
-    # If you are using version 1.x, the "features" block is not allowed.
-    version = "~>2.0"
-    features {}
+terraform {
```
Comment (hjhsalo) [inline main.tf:1]: Maybe move the `required_providers` into the `terraform` block defined at the end of the file.

**Comment by laavamara** [comment]
Done both

**Comment by hjhsalo** [comment]
LGTM
Minor indentation issues, but let's find a Terraform / HCL linter to take of code style differences.

**Pull request merged**